File: src/main/kotlin/fr/jhelp/kotlinLightSamples/PromiseSample.kt
package fr.jhelp.kotlinLightSamples

import fr.jhelp.kotlinLigthTools.promise.Promise

fun promiseTest()
{
    val promise = Promise<Int>()
    promise.getFuture()
        .andThen { number -> increment(number) }
        .andThen { number -> twice(number) }
        .andThen { number -> println("Result : $number") }
    promise.result(20)
}

private fun increment(number: Int): Int
{
    println("Number received : $number")
    return number + 1
}

private fun twice(number: Int): Int
{
    println("Number incremented $number")
    return number * 2
}

File: src/main/swift/Sources/swift/fr/jhelp/kotlinLightSamples/PromiseSample.swift


func promiseTest()
{
    let promise = Promise<Int>()
    promise.getFuture()
        .andThen { number in increment(number) }
        .andThen { number in twice(number) }
        .andThen { number in print("Result : \(number)") }
    promise.result(20)
}

private func increment(_ number: Int) -> Int
{
    print("Number received : \(number)")
    return number + 1
}

private func twice(_ number: Int) -> Int
{
    print("Number incremented \(number)")
    return number * 2
}
